### DISCUSSION TITLE
Amiga Assembly version 1.1.0

### DISCUSSION THREAD

**Discussion by prb28** [prb28/vscode-amiga-assembly#177] [General]
Pre-release version for tests.

<hr /><em>This discussion was created from the release <a href='https://github.com/prb28/vscode-amiga-assembly/releases/tag/pre_1.1.0'>Amiga Assembly</a>.</em>

**Comment by prb28** [comment]
The pre-release can be installed directly from vscode. Select install other version.
![image](https://user-images.githubusercontent.com/10462286/147797657-3935e746-589b-4d52-96d6-61a63b642186.png)

**Comment by phaze101** [comment]
@prb28 tried but the code failed to assemble
![image](https://user-images.githubusercontent.com/50155919/147851102-3acef657-17a7-4cbc-b5d6-166ebe0fccf3.png)

**Reply by prb28** [reply]
Do you have a .vscode/settings.json with the binaries path ?
If yes, you should remove it because the extension will get the 1.1.0 binaries.

**Comment by prb28** [comment]
Documentation for the watch breakpoints:
https://github.com/prb28/vscode-amiga-assembly/wiki/Debugging-emulator#add-a-memory-watch-breakpoint-on-winuae

**Reply by phaze101** [reply]
@prb28 Yes that worked. Will this be needed to be done each time there is an update to the extension when you have a source created with an earlier extension?
Thank you for the documentation of break points that was easy to follow.
Is it possible to have the Symbols to be separate from variables and also somehow sorted so that it is easy to find? Right now they appear as in code, although this is type of sort is also good to have. Maybe one can have different ways to sort.

**Reply by prb28** [reply]
No this .vscode/settings.json value that you had is from my beta version of the 1.0. In the release version it is only needed if you want to override this path.

For the new watch breakpoints, it's managed by vscode. I don't think it's possible to change the order.

**Reply by phaze101** [reply]
@prb28 It is not the Breakpoints I was referring to but the Symbols in the variables area. Is this also managed by VC? The symbols you are breakpoints to them.

**Reply by prb28** [reply]
Ok I understand.
Sorting, it should be possible and it's a good idea (better than the reading order).
Putting it in another treeview (new frame)... I think not because we will loose the ability to add a breakpoint.

**Reply by phaze101** [reply]
@prb28 ok perfect. Sorting is what I was really after.

**Comment by prb28** [comment]
On this version there are great enhancements in completion done by @grahambates, check it out!

**Reply by phaze101** [reply]
@prb28 what are teh new enhancements. Can you list them here

**Reply by prb28** [reply]
- [Feat #184](https://github.com/prb28/vscode-amiga-assembly/pull/184) *grahambates* Completion improvements
- [Feat #183](https://github.com/prb28/vscode-amiga-assembly/pull/183) *grahambates* Preserve case of completions
- [Feat #182](https://github.com/prb28/vscode-amiga-assembly/pull/182) *grahambates* Add documentation for compiler directives
- [Feat #181](https://github.com/prb28/vscode-amiga-assembly/pull/181) *grahambates* Provide completion for labels
- [Issue #179](https://github.com/prb28/vscode-amiga-assembly/issues/179) *grahambates* Labels containing opcodes / control words are formatted incorrect
- [Issue #178](https://github.com/prb28/vscode-amiga-assembly/issues/178) Create ADF with defined Bootblock not work
- [Feat #176](https://github.com/prb28/vscode-amiga-assembly/issues/176) Compress the extension with webpack
- [Pull #173](https://github.com/prb28/vscode-amiga-assembly/pull/173) *dansalvato* Data Generator minor fixes
- [Feat #165](https://github.com/prb28/vscode-amiga-assembly/issues/165) Memory read/write/readwrite watch breakpoint on WinUAE

**Reply by phaze101** [reply]
@prb28 wow thank you for the info @grahambates my compliments. Good job.

**Comment by prb28** [comment]
pre-release 1.1.2:
### [1.1.x] - 
- [Fix #190](https://github.com/prb28/vscode-amiga-assembly/pull/190) *grahambates* Don't complete prefixed local labels
- [Feat #189](https://github.com/prb28/vscode-amiga-assembly/pull/189) *grahambates* Extract docs from comments
- [Feat #187](https://github.com/prb28/vscode-amiga-assembly/pull/187) *grahambates* Improve include path completions
- [Feat #186](https://github.com/prb28/vscode-amiga-assembly/pull/186) *grahambates* Handle local labels correctly
- [Feat #185](https://github.com/prb28/vscode-amiga-assembly/pull/185) *grahambates* Macro and CPU register completions 
- [Feat #184](https://github.com/prb28/vscode-amiga-assembly/pull/184) *grahambates* Completion improvements
- [Feat #183](https://github.com/prb28/vscode-amiga-assembly/pull/183) *grahambates* Preserve case of completions
- [Feat #182](https://github.com/prb28/vscode-amiga-assembly/pull/182) *grahambates* Add documentation for compiler directives
- [Feat #181](https://github.com/prb28/vscode-amiga-assembly/pull/181) *grahambates* Provide completion for labels
- [Issue #179](https://github.com/prb28/vscode-amiga-assembly/issues/179) *grahambates* Labels containing opcodes / control words are formatted incorrect
- [Issue #178](https://github.com/prb28/vscode-amiga-assembly/issues/178) Create ADF with defined Bootblock not work
- [Feat #176](https://github.com/prb28/vscode-amiga-assembly/issues/176) Compress the extension with webpack
- [Pull #173](https://github.com/prb28/vscode-amiga-assembly/pull/173) *dansalvato* Data Generator minor fixes
- [Feat #165](https://github.com/prb28/vscode-amiga-assembly/issues/165) Memory read/write/readwrite watch breakpoint on WinUAE

**Reply by phaze101** [reply]
@prb28 @grahambates wow thanks to both.

@prb28 I have a video which I haven't released yet but would like you to check it out. It is about Debugging using the extension. Drop me an email (prince AT phaze101.com) and I can send you a link to it since I do not wish to share it here yet.

**Comment by prb28** [comment]
- [Feat #192](https://github.com/prb28/vscode-amiga-assembly/pull/192) *grahambates* Improve outline by setting label ranges
- [Feat #191](https://github.com/prb28/vscode-amiga-assembly/pull/191) *grahambates* Add xref symbols for completion/hover
- [Fix #190](https://github.com/prb28/vscode-amiga-assembly/pull/190) *grahambates* Don't complete prefixed local labels
- [Feat #189](https://github.com/prb28/vscode-amiga-assembly/pull/189) *grahambates* Extract docs from comments
- [Fix #188](https://github.com/prb28/vscode-amiga-assembly/pull/188) amigaassembly: compile current file does not honor vasm settings
- [Feat #187](https://github.com/prb28/vscode-amiga-assembly/pull/187) *grahambates* Improve include path completions
- [Feat #186](https://github.com/prb28/vscode-amiga-assembly/pull/186) *grahambates* Handle local labels correctly
- [Feat #185](https://github.com/prb28/vscode-amiga-assembly/pull/185) *grahambates* Macro and CPU register completions 
- [Feat #184](https://github.com/prb28/vscode-amiga-assembly/pull/184) *grahambates* Completion improvements
- [Feat #183](https://github.com/prb28/vscode-amiga-assembly/pull/183) *grahambates* Preserve case of completions
- [Feat #182](https://github.com/prb28/vscode-amiga-assembly/pull/182) *grahambates* Add documentation for compiler directives
- [Feat #181](https://github.com/prb28/vscode-amiga-assembly/pull/181) *grahambates* Provide completion for labels
- [Issue #179](https://github.com/prb28/vscode-amiga-assembly/issues/179) *grahambates* Labels containing opcodes / control words are formatted incorrect
- [Issue #178](https://github.com/prb28/vscode-amiga-assembly/issues/178) Create ADF with defined Bootblock not work
- [Feat #176](https://github.com/prb28/vscode-amiga-assembly/issues/176) Compress the extension with webpack
- [Pull #173](https://github.com/prb28/vscode-amiga-assembly/pull/173) *dansalvato* Data Generator minor fixes
- [Feat #165](https://github.com/prb28/vscode-amiga-assembly/issues/165) Memory read/write/readwrite watch breakpoint on WinUAE

+ data breakpoints persistence.

**Comment by prb28** [comment]
1.1.5 : near a stable 1.2.0 release
- Fix preventing program name for example workspace with ".s"
- Updated documentation
- [Feat #192](https://github.com/prb28/vscode-amiga-assembly/pull/192) *grahambates* Improve outline by setting label ranges
- [Feat #191](https://github.com/prb28/vscode-amiga-assembly/pull/191) *grahambates* Add xref symbols for completion/hover
- [Fix #190](https://github.com/prb28/vscode-amiga-assembly/pull/190) *grahambates* Don't complete prefixed local labels
- [Feat #189](https://github.com/prb28/vscode-amiga-assembly/pull/189) *grahambates* Extract docs from comments
- [Fix #188](https://github.com/prb28/vscode-amiga-assembly/pull/188) amigaassembly: compile current file does not honor vasm settings
- [Feat #187](https://github.com/prb28/vscode-amiga-assembly/pull/187) *grahambates* Improve include path completions
- [Feat #186](https://github.com/prb28/vscode-amiga-assembly/pull/186) *grahambates* Handle local labels correctly
- [Feat #185](https://github.com/prb28/vscode-amiga-assembly/pull/185) *grahambates* Macro and CPU register completions 
- [Feat #184](https://github.com/prb28/vscode-amiga-assembly/pull/184) *grahambates* Completion improvements
- [Feat #183](https://github.com/prb28/vscode-amiga-assembly/pull/183) *grahambates* Preserve case of completions
- [Feat #182](https://github.com/prb28/vscode-amiga-assembly/pull/182) *grahambates* Add documentation for compiler directives
- [Feat #181](https://github.com/prb28/vscode-amiga-assembly/pull/181) *grahambates* Provide completion for labels
- [Issue #179](https://github.com/prb28/vscode-amiga-assembly/issues/179) *grahambates* Labels containing opcodes / control words are formatted incorrect
- [Issue #178](https://github.com/prb28/vscode-amiga-assembly/issues/178) Create ADF with defined Bootblock not work
- [Feat #176](https://github.com/prb28/vscode-amiga-assembly/issues/176) Compress the extension with webpack
- [Pull #173](https://github.com/prb28/vscode-amiga-assembly/pull/173) *dansalvato* Data Generator minor fixes
- [Feat #165](https://github.com/prb28/vscode-amiga-assembly/issues/165) Memory read/write/readwrite watch breakpoint on WinUAE

**Comment by prb28** [comment]
- [Fix #200](https://github.com/prb28/vscode-amiga-assembly/issues/200) Undefined symbols not reported in errors view
- [Fix #197](https://github.com/prb28/vscode-amiga-assembly/issues/197) File resolution of the build errors wrong
- Fix preventing program name for example workspace with ".s"
- Updated documentation
- [Feat #192](https://github.com/prb28/vscode-amiga-assembly/pull/192) *grahambates* Improve outline by setting label ranges
- [Feat #191](https://github.com/prb28/vscode-amiga-assembly/pull/191) *grahambates* Add xref symbols for completion/hover
- [Fix #190](https://github.com/prb28/vscode-amiga-assembly/pull/190) *grahambates* Don't complete prefixed local labels
- [Feat #189](https://github.com/prb28/vscode-amiga-assembly/pull/189) *grahambates* Extract docs from comments
- [Fix #188](https://github.com/prb28/vscode-amiga-assembly/pull/188) amigaassembly: compile current file does not honor vasm settings
- [Feat #187](https://github.com/prb28/vscode-amiga-assembly/pull/187) *grahambates* Improve include path completions
- [Feat #186](https://github.com/prb28/vscode-amiga-assembly/pull/186) *grahambates* Handle local labels correctly
- [Feat #185](https://github.com/prb28/vscode-amiga-assembly/pull/185) *grahambates* Macro and CPU register completions 
- [Feat #184](https://github.com/prb28/vscode-amiga-assembly/pull/184) *grahambates* Completion improvements
- [Feat #183](https://github.com/prb28/vscode-amiga-assembly/pull/183) *grahambates* Preserve case of completions
- [Feat #182](https://github.com/prb28/vscode-amiga-assembly/pull/182) *grahambates* Add documentation for compiler directives
- [Feat #181](https://github.com/prb28/vscode-amiga-assembly/pull/181) *grahambates* Provide completion for labels
- [Issue #179](https://github.com/prb28/vscode-amiga-assembly/issues/179) *grahambates* Labels containing opcodes / control words are formatted incorrect
- [Issue #178](https://github.com/prb28/vscode-amiga-assembly/issues/178) Create ADF with defined Bootblock not work
- [Feat #176](https://github.com/prb28/vscode-amiga-assembly/issues/176) Compress the extension with webpack
- [Pull #173](https://github.com/prb28/vscode-amiga-assembly/pull/173) *dansalvato* Data Generator minor fixes
- [Feat #165](https://github.com/prb28/vscode-amiga-assembly/issues/165) Memory read/write/readwrite watch breakpoint on WinUAE

**Reply by prb28** [reply]
version 1.1.6 pre-release
